src/main/java/com/google/devtools/build/lib/query2/cquery/LabelAndConfigurationOutputFormatterCallback.java

```diff
@@ -24,33 +24,46 @@
 
 /** Default Output callback for cquery. Prints a label and configuration pair per result. */
 public class LabelAndConfigurationOutputFormatterCallback extends CqueryThreadsafeCallback {
-
+  private final boolean showKind;
   public LabelAndConfigurationOutputFormatterCallback(
       ExtendedEventHandler eventHandler,
       CqueryOptions options,
       OutputStream out,
       SkyframeExecutor skyframeExecutor,
-      TargetAccessor<ConfiguredTarget> accessor) {
-    super(eventHandler, options, out, skyframeExecutor, accessor);
+      TargetAccessor<ConfiguredTarget> accessor,
+      ConfiguredTargetAccessor configuredTargetAccessor,
+      boolean showKind) {
+    super(eventHandler, options, out, skyframeExecutor, accessor, configuredTargetAccessor, showKind);
   }
 
+
   @Override
   public String getName() {
-    return "label";
+    return showKind ? "label_kind" : "label";
   }
 
   @Override
   public void processOutput(Iterable<ConfiguredTarget> partialResult) {
     for (ConfiguredTarget configuredTarget : partialResult) {
       BuildConfiguration config =
           skyframeExecutor.getConfiguration(eventHandler, configuredTarget.getConfigurationKey());
-
-      StringBuilder output =
-          new StringBuilder()
-              .append(configuredTarget.getLabel())
-              .append(" (")
-              .append(config != null && config.isHostConfiguration() ? "HOST" : config)
-              .append(")");
+      StringBuilder output;
+      if(showKind){
+            output =
+            new StringBuilder()
+                .append(configuredTargetAccessor.getTargetFromConfiguredTarget(configuredTarget))
```

Comment:
Following my longer comment: just change this to `accessor.getTargetKind(configuredTarget)` and I think it'll just work.